pkg/skaffold/runner/label/label.go

```diff
@@ -86,67 +90,81 @@ func updateRuntimeObject(client clientgo.Interface, labels map[string]string, re
-		_, err = sets.UpdateStatus(s)
+		apiObject := modifiedObj.(*appsv1.StatefulSet)
+		addSkaffoldLabels(labels, &apiObject.ObjectMeta)
+		modifiedJSON, _ := json.Marshal(modifiedObj)
+		p, _ := patch.CreateTwoWayMergePatch(originalJSON, modifiedJSON, apiObject)
+		_, err = client.AppsV1().StatefulSets(retrieveNamespace(res.Namespace, apiObject.ObjectMeta)).Patch(apiObject.Name, types.StrategicMergePatchType, p)
 	case *appsv1beta1.StatefulSet:
-		s := obj.(*appsv1beta1.StatefulSet)
-		addSkaffoldLabels(labels, &s.ObjectMeta)
-		sets := client.AppsV1beta1().StatefulSets(retrieveNamespace(res.Namespace, s.ObjectMeta))
-		_, err = sets.UpdateStatus(s)
+		apiObject := modifiedObj.(*appsv1beta1.StatefulSet)
+		addSkaffoldLabels(labels, &apiObject.ObjectMeta)
+		modifiedJSON, _ := json.Marshal(modifiedObj)
+		p, _ := patch.CreateTwoWayMergePatch(originalJSON, modifiedJSON, apiObject)
+		_, err = client.AppsV1beta1().StatefulSets(retrieveNamespace(res.Namespace, apiObject.ObjectMeta)).Patch(apiObject.Name, types.StrategicMergePatchType, p)
 	case *appsv1beta2.StatefulSet:
-		s := obj.(*appsv1beta2.StatefulSet)
-		addSkaffoldLabels(labels, &s.ObjectMeta)
-		sets := client.AppsV1beta2().StatefulSets(retrieveNamespace(res.Namespace, s.ObjectMeta))
-		_, err = sets.UpdateStatus(s)
+		apiObject := modifiedObj.(*appsv1beta2.StatefulSet)
+		addSkaffoldLabels(labels, &apiObject.ObjectMeta)
+		modifiedJSON, _ := json.Marshal(modifiedObj)
+		p, _ := patch.CreateTwoWayMergePatch(originalJSON, modifiedJSON, apiObject)
+		_, err = client.AppsV1beta2().StatefulSets(retrieveNamespace(res.Namespace, apiObject.ObjectMeta)).Patch(apiObject.Name, types.StrategicMergePatchType, p)
 	case *extensionsv1beta1.DaemonSet:
-		s := obj.(*extensionsv1beta1.DaemonSet)
-		addSkaffoldLabels(labels, &s.ObjectMeta)
-		sets := client.ExtensionsV1beta1().DaemonSets(retrieveNamespace(res.Namespace, s.ObjectMeta))
-		_, err = sets.UpdateStatus(s)
+		apiObject := modifiedObj.(*extensionsv1beta1.DaemonSet)
+		addSkaffoldLabels(labels, &apiObject.ObjectMeta)
+		modifiedJSON, _ := json.Marshal(modifiedObj)
+		p, _ := patch.CreateTwoWayMergePatch(originalJSON, modifiedJSON, apiObject)
+		_, err = client.ExtensionsV1beta1().DaemonSets(retrieveNamespace(res.Namespace, apiObject.ObjectMeta)).Patch(apiObject.Name, types.StrategicMergePatchType, p)
 	case *appsv1.ReplicaSet:
-		rs := obj.(*appsv1.ReplicaSet)
-		addSkaffoldLabels(labels, &rs.ObjectMeta)
-		sets := client.AppsV1().ReplicaSets(retrieveNamespace(res.Namespace, rs.ObjectMeta))
-		_, err = sets.UpdateStatus(rs)
+		apiObject := modifiedObj.(*appsv1.ReplicaSet)
+		addSkaffoldLabels(labels, &apiObject.ObjectMeta)
+		modifiedJSON, _ := json.Marshal(modifiedObj)
+		p, _ := patch.CreateTwoWayMergePatch(originalJSON, modifiedJSON, apiObject)
+		_, err = client.AppsV1().ReplicaSets(retrieveNamespace(res.Namespace, apiObject.ObjectMeta)).Patch(apiObject.Name, types.StrategicMergePatchType, p)
```

Comment:
What was the problem using the dynamic client here? I thought that was working?

Reply:
some of it works, but there are a few types that the version I'm using needs that are incompatible with other dependency versions we already have vendored. tl;dr it's a dep nightmare, I'll work through it eventually